MAD.Integration.TableauCRM/Services/QueryClient.cs

```diff
@@ -0,0 +1,104 @@
+﻿using MAD.Integration.TableauCRM.Data;
+using SqlKata.Execution;
+
+namespace MAD.Integration.TableauCRM.Services
+{
+    public class QueryClient : IQueryClient
+    {
+        private readonly IQueryFactoryFactory queryFactoryFactory;
+
+        public QueryClient(IQueryFactoryFactory queryFactoryFactory)
+        {
+            this.queryFactoryFactory = queryFactoryFactory;
+        }
+
+        public async Task<IEnumerable<dynamic>> QuerySourceTable(Configuration configuration)
+        {
+            using var db = this.queryFactoryFactory.Create(configuration.DatabaseName);
+
+            // Query the configured database/table name and return the results
+            return await db.Query(configuration.TableName).GetAsync<dynamic>();
+        }
+
+        public async Task<IEnumerable<ColumnDefinition>> GetSourceTableColumns(Configuration configuration)
+        {
+            using var db = this.queryFactoryFactory.Create(configuration.DatabaseName);
+
+            //Retrieve the columns with their type, precision and scale for the input table
+            var rawQuery = $@"
```

Comment:
Consider using exec sp_describe_first_result_set @tsql as this will allow us to support views and stored procs

Reply:
https://dev.azure.com/maitdev/MAD.Integration.FlatFile/_git/MAD.Integration.FlatFile?path=/MAD.Integration.FlatFile/Services/SqlResultSetFactory.cs

Reply:
Added Result Set functionality similar to MAD.Integration.FlatFile